actionview/test/ujs/public/test/data-remote.js

```diff
@@ -82,6 +82,20 @@ asyncTest('right/mouse-wheel-clicking on a link does not fire ajaxyness', 0, fun
   setTimeout(function() { start() }, 13)
 })
 
+asyncTest('clicking on a link via a non-mouse Event (such as from js) works ', 1, function() {
+  var link = $('a[data-remote]')
+
+  link
+    .removeAttr('data-params')
+    .bindNative('ajax:beforeSend', function() {
+      ok(true, 'ajax should be triggered')
+    })
+
+  link.triggerNative('jsclick')
```

Comment:
You can avoid this and also make sure Rails.fire() works directly if you trigger using
```suggestion
  Rails.fire(link[0], 'click')
```

Reply:
Beautiful, thank you. I didn't think `Rails.fire` would be callable from the test, that's perfect!